execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -294,6 +295,90 @@ where
     Ok(missing_descendants)
 }
 
+#[cfg(test)]
+pub fn check_integrity<K, V, T, S, E>(
+    _correlation_id: CorrelationId,
+    txn: &T,
+    store: &S,
+    trie_keys_to_visit: Vec<Blake2bHash>,
+) -> Result<(), E>
+where
+    K: ToBytes + FromBytes + Eq + std::fmt::Debug,
+    V: ToBytes + FromBytes + std::fmt::Debug,
+    T: Readable<Handle = S::Handle>,
+    S: TrieStore<K, V>,
+    S::Error: From<T::Error>,
+    E: From<S::Error> + From<bytesrepr::Error>,
+{
+    for state_root in &trie_keys_to_visit {
+        match store.get(txn, &state_root)? {
+            Some(Trie::Node { .. }) => {}
+            _ => panic!("Should have a pointer block node as state root"),
```

Comment:
The state root `Trie` value is special in the sense that it is always a `Trie::Node`, so, unless I'm misundertanding something, we need to do that check here.

Reply:
I mean we call `store.get` twice for all initial entries in `trie_keys_to_visit`.
And aren't the initial entries exactly the ones with an empty path? (The roots?)

If we did this check below, inside the `while` loop, we wouldn't need this `get` call here: Below, we'd check that if `path.is_empty()` then it's a `Trie::Node`, otherwise panic.